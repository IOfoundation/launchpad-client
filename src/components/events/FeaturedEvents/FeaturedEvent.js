import React from 'react';
import PropTypes from 'prop-types';

const FeaturedEvent = props => {
  return (
    <div className="featured-event">
      <h3 className="featured-event__title text-bold">{props.title}</h3>
      <p className="featured-event__description">{props.description}</p>
      <p className="featured-event__information">
        <span className="featured-event__information__date text-bold">
          {props.date}
        </span>{' '}
        {'by'} <span>{props.name}</span>
      </p>
    </div>
  );
};

FeaturedEvent.propTypes = {
  date: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default FeaturedEvent;
