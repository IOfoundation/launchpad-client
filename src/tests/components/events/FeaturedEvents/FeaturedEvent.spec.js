import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FeaturedEvent from '../../../../components/events/FeaturedEvents/FeaturedEvent';

const props = {
  date: 'date',
  description: 'description',
  name: 'name',
  title: 'title:',
};

describe('<FeaturedEvent />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FeaturedEvent {...props} />);
  });

  it('renders snapshot of FeaturedEvent', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
