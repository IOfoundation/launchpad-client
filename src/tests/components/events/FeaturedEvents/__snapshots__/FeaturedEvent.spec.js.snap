// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedEvent /> renders snapshot of FeaturedEvent 1`] = `
<div
  className="featured-event"
>
  <h3
    className="featured-event__title text-bold"
  >
    title:
  </h3>
  <p
    className="featured-event__description"
  >
    description
  </p>
  <p
    className="featured-event__information"
  >
    <span
      className="featured-event__information__date text-bold"
    >
      date
    </span>
     
    by
     
    <span>
      name
    </span>
  </p>
</div>
`;
